### PR TITLE
Add Neon Postgres vector database OpenAI cookbook

### PR DESCRIPTION
**Pull request by danieltprice**
This PR adds OpenAI cookbook examples for Neon Postgres as a vector database. The examples demonstrate how to use Neon Postgres with OpenAI and the pgvector extension for vector similarity search. The pgvector open-source extension enables Postgres as a vector store.

### CODE REVIEW

File: examples/vector_databases/neon/neon-postgres-vector-search-pgvector.ipynb [hunk 1]
```diff
@@ -0,0 +1,538 @@
...
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Vector similarity search using Neon Postgres with pgvector and OpenAI\n",
+    "\n",
+    "This notebook guides you through using [Neon Serverless Postgres](https://neon.tech/) as a vector database for OpenAI embeddings. It demonstrates how to:\n",
+    "\n",
+    "1. Use embeddings created by OpenAI API\n",
```
Comment (simonpfish) [inline examples/vector_databases/neon/neon-postgres-vector-search-pgvector.ipynb:12]: could add periods here for consistency with the other lists

File: examples/vector_databases/neon/neon-postgres-vector-search-pgvector.ipynb [hunk 2]
```diff
@@ -0,0 +1,538 @@
...
+     ]
+    }
+   ],
+   "source": [
+    "# Test to ensure that your OpenAI API key is defined as an environment variable\n",
+    "# If you run this notebook locally, you may have to reload the terminal and the notebook to make the environment available\n",
+    "\n",
+    "import os\n",
+    "\n",
+    "if os.getenv(\"OPENAI_API_KEY\") is not None:\n",
```
Comment (simonpfish) [inline examples/vector_databases/neon/neon-postgres-vector-search-pgvector.ipynb:95]: alternatively, you could use `getpass` for an easier way to get the secret if it's not defined in the env

**Review by simonpfish — APPROVED** [review]
This is great, thank you for the contribution!

File: examples/vector_databases/neon/neon-postgres-vector-search-pgvector.ipynb [hunk 3]
```diff
@@ -0,0 +1,538 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Vector similarity search using Neon Postgres with pgvector and OpenAI\n",
```
Comment (simonpfish) [inline examples/vector_databases/neon/neon-postgres-vector-search-pgvector.ipynb:8]: Simpler title:

```suggestion
    "# Vector similarity search using Neon Postgres\n",
```

File: examples/vector_databases/neon/neon-postgres-vector-search-pgvector.ipynb (hunk 2 above)
Comment (danieltprice) [reply examples/vector_databases/neon/neon-postgres-vector-search-pgvector.ipynb:97]: Added a getpass option that can be used if an environment variable is not set.

File: examples/vector_databases/neon/neon-postgres-vector-search-pgvector.ipynb (hunk 1 above)
Comment (danieltprice) [reply examples/vector_databases/neon/neon-postgres-vector-search-pgvector.ipynb]: Thank you, @simonpfish. Added periods, as suggested.

**Pull request merged**